src/storage/hummock_sdk/src/filter_key_extractor.rs

```diff
@@ -56,8 +59,24 @@ impl FilterKeyExtractorImpl {
             .map(|col_order| col_order.index as usize)
             .collect();
 
-        let match_read_pattern =
-            !dist_key_indices.is_empty() && pk_indices.starts_with(&dist_key_indices);
+        let dist_key_in_pk_indices = dist_key_indices
+            .iter()
+            .map(|&di| {
+                pk_indices
+                    .iter()
+                    .position(|&pi| di == pi)
+                    .unwrap_or_else(|| {
+                        panic!(
+                            "distribution key {:?} must be a subset of primary key {:?}",
+                            dist_key_indices, pk_indices
+                        )
+                    })
+            })
+            .collect_vec();
+
+        let match_read_pattern = !dist_key_in_pk_indices.is_empty()
```

Comment:
use `get_dist_key_start_index_in_pk`?

Reply:
not resolved?

Reply:
also, per our offline discussion, L68 can be replaced with the `DummyFilterKeyExtractor`.